hub/core/dataset.py

```diff
@@ -60,10 +64,25 @@ def __init__(
             AuthorizationException: If a Hub cloud path (path starting with hub://) is specified and the user doesn't have access to the dataset.
             PathNotEmptyException: If the path to the dataset doesn't contain a Hub dataset and is also not empty.
         """
-        self._read_only = read_only
         # uniquely identifies dataset
         self.path = get_path_from_storage(storage)
         self.storage = storage
+        if read_only:
+            self._read_only = True
+        else:
+            base_storage = get_base_storage(storage)
+            if isinstance(
+                base_storage, S3Provider
+            ):  # Dataset locking only for S3 datasets
```

Comment:
why is dataset locking only for s3 providers?

Reply:
We discussed it during the call. @AbhinavTuli ^^